01-supp-addressingData.Rmd

```diff
@@ -0,0 +1,168 @@
+---
+layout: page
+title: Programming with R
+subtitle: Addressing data
+minutes: 20
+---
+
+
+```{r, include = FALSE}
+source('tools/chunk-options.R')
+```
+
+> ## Objectives {.objectives}
+>
+> Understand the 3 different ways R can address data inside a data frame.
+>
+> Combine different methods for addressing data with the assignment operator to update subsets of data
+
+R is a powerful language for data manipulation. There are 3 main ways for addressing data inside R objects.
+
+* By index (slicing)
+* By logical vector
+* By name (columns and lists only)
+
+Lets start by loading some sample data:
+
+```{r readData}
+dat<-read.csv(file='data/sample.csv',header=TRUE, stringsAsFactors=FALSE)
```

Comment:
What is the general thought about using or not using `stringsAsFactors=FALSE` in these intro lessons?  I can see a number of cases in which is would be quite useful to leave it on.  I also worry that throwing this in there get's into a bit of a tangle, but I suppose the lesson materials are intended for the instructors to teach from and not for the students directly, in which case it doesn't matter all that much I guess.